File: app/src/main/java/com/majinnaibu/monstercards/data/converters/ArmorTypeConverter.java
package com.majinnaibu.monstercards.data.converters;

import androidx.room.TypeConverter;

import com.majinnaibu.monstercards.data.enums.ArmorType;

public class ArmorTypeConverter {

    @TypeConverter
    public static String fromArmorType(ArmorType armorType) {
        return armorType.stringValue;
    }

    @TypeConverter
    public static ArmorType armorTypeFromStringValue(String stringValue) {
        return ArmorType.valueOfString(stringValue);
    }
}

File: app/src/main/java/com/majinnaibu/monstercards/data/converters/ArmorTypeConverter.kt
package com.majinnaibu.monstercards.data.converters

import androidx.room.TypeConverter
import com.majinnaibu.monstercards.data.enums.ArmorType
import com.majinnaibu.monstercards.data.enums.ArmorType.Companion.valueOfString

object ArmorTypeConverter {
    @JvmStatic
    @TypeConverter
    fun fromArmorType(armorType: ArmorType): String {
        return armorType.stringValue
    }

    @JvmStatic
    @TypeConverter
    fun armorTypeFromStringValue(stringValue: String?): ArmorType {
        return valueOfString(stringValue!!)
    }
}